packages/@aws-cdk/cdk/lib/core/util.ts

```diff
@@ -47,4 +49,70 @@ export function ignoreEmpty(o: any): any {
     }
 
     return o;
-}
\ No newline at end of file
+}
+
+/**
+ * Result of the split of a string with Tokens
+ *
+ * Either a literal part of the string, or an unresolved Token.
+ */
+export type MarkerSpan = { type: 'string'; value: string } | { type: 'marker'; id: string };
+
+/**
+ * Split a string up into String Spans and Marker Spans
+ */
+export function splitOnMarkers(s: string, beginMarker: string, idPattern: string, endMarker: string): MarkerSpan[] {
+    // tslint:disable-next-line:max-line-length
+    const re = new RegExp(`${regexQuote(beginMarker)}(${idPattern})${regexQuote(endMarker)}`, 'g');
```

Comment:
Can we just limit what `beginMarker` and `endMarker` accept instead of yet another escaping? There's no real need for those markers to be super flexible, and also, they should be picked based on their aesthetic form when embedded in strings, so picking something that will eventually be escaped kinda defeats the purpose, no?

Reply:
Your previous favorite pick, `${Token[` and `]}` contain 5 characters that need to be escaped when used in a regex. The alternative is to build our markers completely out of alphanumerics, which is not going to improve aesthetics and going to make collissions more likely.

Reply:
Got it. Didn't realize this was in order to formulate the regex. The resulting marker is not going to be escaped. Sorry.